rcl/src/rcl/publisher.c

```diff
@@ -104,6 +104,13 @@ rcl_publisher_init(
   RCL_CHECK_FOR_NULL_WITH_MSG(
     publisher->impl, "allocating memory failed", ret = RCL_RET_BAD_ALLOC; goto cleanup);
 
+  // Cache disable flag to loan messasges on this publisher
+  bool disable_loaned_message = false;
+  ret = rcl_get_disable_loaned_message(&disable_loaned_message);
+  if (ret == RCL_RET_OK && disable_loaned_message) {
+    publisher->disable_loaned_message = true;
+  }
```

Comment:
I think we need to deal with errors better here:

```suggestion
  // Cache disable flag to loan messages on this publisher
  bool disable_loaned_message = false;
  ret = rcl_get_disable_loaned_message(&disable_loaned_message);
  if (ret != RCL_RET_OK) {
    goto cleanup;
  }
  publisher->disable_loaned_message = disable_loaned_message;
```

Same in the subscription below.

Reply:
thanks, this is a bug. i will fix it.